### PR TITLE
shanoir-issue#1317 dataset acquisition update

### LINKED ISSUE #1317: Dataset acquisition edition fails
message: "Failed to evaluate expression 'hasAnyRole('ADMIN', 'EXPERT') and #datasetAcquisitionId == #datasetAcquisition.getId() and @datasetSecurityService.hasUpdateRightOnDatasetAcquisition(#datasetAcquisition, 'CAN_ADMINISTRATE')'"

To be updated

### PR DESCRIPTION
**Pull request by jcomedouteau**
How to test:
- Update a dataset acquisition as an expert, it works.

Closes #1317

### CODE REVIEW

**Comment by Nunien** [comment]
@jcomedouteau test are failing

**Comment by jcomedouteau** [comment]
Yep, I'm on it !

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/security/DatasetSecurityService.java [hunk 1]
```diff
@@ -66,6 +67,9 @@ public class DatasetSecurityService {
 
 	@Autowired
 	StudyRepository studyRepository;
+
+	@Autowired
+	DatasetAcquisitionMapper datasetAcquisitionMapper;
```
Comment (Nunien) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/security/DatasetSecurityService.java:72]: This is not used, is it ?

**Review by Nunien — CHANGES_REQUESTED** [review]

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/security/DatasetSecurityService.java (hunk 1 above)
Comment (jcomedouteau) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/security/DatasetSecurityService.java]: It isn't, I removed it! :+1:

**Comment by jcomedouteau** [comment]
Do you have the "admin" right on this study ?

**Comment by Nunien** [comment]
@jcomedouteau I think you compare Longs with == , maybe try .equals() ?

**Comment by jcomedouteau** [comment]
I don't get it, I removed this line and it still does not work. It seems that 
`return commService.hasRightOnStudy(dataset.getStudyId(), rightStr);
`
returns false..

**Comment by jcomedouteau** [comment]
I'm closing this one as this is not relevant anymore.

**Pull request closed without merging**